src/WebJobs.Script.WebHost/Security/KeyManagement/KeyVaultSecretsRepository.cs

```diff
@@ -195,6 +203,40 @@ private async Task<ScriptSecrets> ReadFunctionSecrets(string functionName)
             return functionSecrets;
         }
 
+        private async Task<List<IPage<SecretItem>>> GetSecretsPagesAsync()
```

Comment:
Can we add some tests for the pagination?

Reply:
E2E tests:
https://github.com/Azure/azure-functions-host/blob/dev/test/WebJobs.Script.Tests.Integration/Host/SecretsRepositoryTests.cs#L50
https://github.com/Azure/azure-functions-host/blob/dev/test/WebJobs.Script.Tests.Integration/Host/SecretsRepositoryTests.cs#L87

Unit tests: (we can add one for FindSecrets)
https://github.com/Azure/azure-functions-host/blob/dev/test/WebJobs.Script.Tests/Security/KeyVaultSecretsRepositoryTests.cs